examples/tests/valid/indexing.test.w

```diff
@@ -0,0 +1,68 @@
+bring cloud;
+bring expect;
+
+let arr = Array<num>[1, 2, 3];
+
+assert(arr[0] == 1);
```

Comment:
Spec update? Doc update would be nice as well, but IMO probably fine to get this PR in first.

Reply:
👍 Added a section to the language reference